File: Leetcode/merge_sorted_array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        if len(nums1) != m+n:
            raise ValueError("nums1 should be longer")
        print(nums1)
        print(nums2)
        for i in range(m, m+n):
            nums1[i] = nums2[i-m]
        nums1.sort()
if __name__ == "__main__":
    task = Solution()
    nums1 = [1,2,3,0,0,0]
    m = 3 
    nums2 = [2,5,6]
    n = 3
    task.merge(nums1, m, nums2, n)   
    print(nums1)

File: Leetcode/88_Merge-sorted-array/merge_sorted_array.cpp
#include <iostream>
#include <vector>
#include <string>
#include<algorithm>
using namespace  std;
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if(nums1.size() != n+m){
            printf("nums1 should be longer");
        }
        for (int i = 0; i < n; i++) {
            nums1[m+i] = nums2[i];
        }
        std::sort(nums1.begin(), nums1.end());


    }
};

int main() {
    Solution task;
    std::vector<int> nums1 = {1,2,3,0,0,0};
    int m = 3;
    int n = 3;
    std::vector<int> nums2 = {2,5,6};

    task.merge(nums1, m, nums2, n);
    for (int i = 0; i < nums1.size(); i++) {
        cout << nums1[i] << " ";
    }
    return 0;    
}